vault/auth/kubernetes/kubernetes.go

```diff
@@ -23,18 +19,23 @@ func GetJWT(serviceAccountName, namespace string) (string, error) {
 		return "", errors.Errorf("No service account secret found")
 	}
 	secretName := serviceAccount.Secrets[0].Name
```

Comment:
```suggestion
	secretName := serviceAccount.Secrets[0].Name
```
Add safety check whether `serviceAccount.Secrets` is empty. If for some reason this `serviceAccount.Secrets` is empty, then it will get panic.